test/repositories/application/ReportingRepoBSONToModelHelperSpec.scala

```diff
@@ -0,0 +1,298 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories.application
+
+import model.ApplicationRoute.{ apply => _ }
+import model.ProgressStatuses.{ PHASE1_TESTS_PASSED => _, SUBMITTED => _ }
+import model.report.{ VideoInterviewQuestionTestResult, VideoInterviewTestResult }
+import reactivemongo.bson.{ BSONArray, BSONDateTime, BSONDocument }
+import testkit.UnitWithAppSpec
+
+class ReportingRepoBSONToModelHelperSpec extends UnitWithAppSpec {
+
+  def bsonToModelHelper = ReportingRepoBSONToModelHelper
+
+  "toPhase3TestResults" should {
+    "return corresponding VideoInterviewTestResult when only one reviewer and one reviewed callback" in new Fixture {
+      val videoInterviewTestResult = bsonToModelHelper.toPhase3TestResults(Some(oneReviewedOneReviewerReviewedPhase3BSONDoc(1.5)))
+      videoInterviewTestResult mustBe
+        Some(expectedBaseResult.copy(question1 = VideoInterviewQuestionTestResult(Some(1.5), Some(2.5)), overallTotal = 39.5)
+      )
+    }
+
+    "return corresponding VideoInterviewTestResult when only two reviewer and one reviewed callback" in new Fixture {
+      val videoInterviewTestResult = bsonToModelHelper.toPhase3TestResults(Some(oneReviewedTwoReviewerReviewedPhase3BSONDoc(2.5, 2.5)))
+      videoInterviewTestResult mustBe
+        Some(expectedBaseResult.copy(question1 = VideoInterviewQuestionTestResult(Some(2.5), Some(2.5)), overallTotal = 40.5)
+      )
+    }
+
+    "return corresponding VideoInterviewTestResult when two reviewer and three reviewed callback" in new Fixture {
+      val videoInterviewTestResult =
+        bsonToModelHelper.toPhase3TestResults(Some(threeReviewedTwoReviewerReviewedPhase3BSONDoc(2.5, 2.5, 3.5, 1.0, 1.0, 4.0)))
+      videoInterviewTestResult mustBe
+        Some(expectedBaseResult.copy(question1 = VideoInterviewQuestionTestResult(Some(1.0), Some(2.5)), overallTotal = 39.0)
+        )
+    }
+  }
+
+  trait Fixture {
+    val baseTotalOverall = 38.0
+    val expectedBaseResult = VideoInterviewTestResult(
+      VideoInterviewQuestionTestResult(None, Some(2.5)),
+      VideoInterviewQuestionTestResult(Some(1.5), Some(2.0)),
+      VideoInterviewQuestionTestResult(Some(3.0), Some(2.5)),
+      VideoInterviewQuestionTestResult(Some(3.5), Some(1.0)),
+      VideoInterviewQuestionTestResult(Some(3.5), Some(2.0)),
+      VideoInterviewQuestionTestResult(Some(4.0), Some(2.5)),
+      VideoInterviewQuestionTestResult(Some(3.5), Some(4.0)),
+      VideoInterviewQuestionTestResult(Some(1.0), Some(1.5)),
+      baseTotalOverall
+    )
+
+
+    //scalastyle:off method.length
+    // overall should be 38 + score
+    def reviewerBSONDoc(score: Double) = BSONDocument(
+      "name" -> "Garfiled the cat",
+      "email" -> "garfieldthecat@mailinator.com",
```

Comment:
How can I know if something exists or not? In any case any domain will eventually exist at some point in the future. Can you give an example of a valid one?